frontend/src/components/result/VoteRestaurantTable.tsx

```diff
@@ -0,0 +1,168 @@
+import {Button, CardContent, TextField} from '@mui/material';
+import Card from '@mui/material/Card';
+import {DataCompleteDateInfos, transferUserTimeVote, UserTimeVote} from "../../service/tastydate-api-service";
+import "./Tables.scss"
+import Checkbox from '@mui/material/Checkbox';
+import {useContext, useState} from "react";
+import {AuthContext} from "../../context/AuthProvider";
+import CheckIcon from '@mui/icons-material/Check';
+import CloseIcon from '@mui/icons-material/Close';
+import React from 'react';
+
+interface VoteRestaurantTableProps {
+    transferSettingsItem: DataCompleteDateInfos
+}
+
+export default function VoteRestaurantTable({transferSettingsItem}: VoteRestaurantTableProps) {
+
+    const {token} = useContext(AuthContext);
+
+    const [userName, setUserName] = useState<string>("");
+        const [checked, setChecked] = useState<boolean[]>([false]);
+
+        const [rowUserTimeVote, setRowUserTimeVote] = useState<UserTimeVote[]>([]);
+
+        const handleChange = (index: number) => {
+            const newChecked=[...checked];
+            newChecked[index]=!newChecked[index];
+            setChecked(newChecked);
+            }
+
+            const addUserVote = () => {
+                if (userName !== "") {
+                    const newUserVote = {
+                        displayedName: userName,
+                        votedTimes: checked
+                    }
+                    transferUserTimeVote(newUserVote, token)
+                        .then((response) => {
+                            setRowUserTimeVote([...rowUserTimeVote, response.data])
+                        })
+                        .catch((err) => {
+                            console.error(err.message);
+                        })
+                }
+            }
+
+            if (rowUserTimeVote.length === 0) {
+    return (
```

Comment:
Im not sure where this two cases differentiate? But I would extract them into one or two seperate components.

Reply:
Thanks for your notice. It was a pilot version, just that I could execute the code. But as you marked correctly, this code was not necessary and I removed it.